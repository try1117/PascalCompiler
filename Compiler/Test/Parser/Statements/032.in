program test;
var
  i: integer;
  arr: array[1..10] of integer;
begin
  if (arr[5]) then i := i;
  read(arr[1], arr[2], arr[3]);
  for i := 1 to 10 do read(arr[i]);
end.
