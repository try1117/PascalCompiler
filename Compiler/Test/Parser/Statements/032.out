test : function()
   resultType : Nil

test declarations:
   i : Integer

   arr : Array [1, 10] of Integer

|-- Statements
|            |-- If
|            |    |-- []
|            |    |    |-- arr
|            |    |    --- 5
|            |    --- :=
|            |         |-- i
|            |         --- i
|            |-- Read
|            |      |-- []
|            |      |    |-- arr
|            |      |    --- 1
|            |      |-- []
|            |      |    |-- arr
|            |      |    --- 2
|            |      --- []
|            |           |-- arr
|            |           --- 3
|            --- For
|                  |-- i
|                  |-- 1
|                  |-- 10
|                  --- Read
|                         |-- []
|                         |    |-- arr
|                         |    --- i

